SNP summary statistics and grid view

gijskant opened the pull request.

rnugraha committed:
fix initial autocompletion problem and refactoring

rnugraha committed:
Refactor autocomplete to have its own service

gijskant committed:
Attempt at summary + grid view for SNP data.

gijskant committed:
Attempt at backend for summary + grid view for SNP data.

gijskant committed:
Ad hoc frontend code for SNP filter selection.

rnugraha committed:
Merge branch 'snp-summary-grid' into refactor_highdim_dialog

gijskant committed:
Fixed summary statistics for SNP data.

- Compute actual counts for selected subsets;
- Sort genotypes;
- Split plot into two separate plots for the subsets;

rnugraha committed:
Merge branch 'refactor_highdim_dialog' into snp-summary-grid-ui

Conflicts:
	grails-app/services/ChartService.groovy
	grails-app/views/chart/_highdimComparison.gsp
	web-app/css/datasetExplorer.css
	web-app/js/datasetExplorer/datasetExplorer.js

rnugraha committed:
Enable snp id filtering in summary statistic

rnugraha committed:
removed high dimensional details ajax call since it is not needed here

rnugraha committed:
Add loading animation when user drop HD into summary stat

gijskant committed:
Fixed concept filtering for SNP summary + grid.

gijskant committed:
Remove double plot; fix CSS for SNP summary plot.

gijskant commented:
Summary statistics and grid view for SNP data: select a study (or subset(s)) with SNP data (e.g., Bio Bank\CARDS), go to Summary Statistics and drap in a concept with SNP data (e.g., Genotyping\Perlegen_600k\Blood) and select SNPs in the filter popup (e.g., SNP name 'rs12890222').

JanKanis commented on grails-app/controllers/ChartController.groovy:
```diff
@@ -187,8 +187,17 @@ class ChartController {
         // We add the key to our cache set
         chartService.keyCache.add(concept)
 
+        // FIXME: get filters from concept data
```
Why wasn't this fixed then?

gijskant replied:
I now removed the comment, we approached it differently than in the data export screen (where we do combine filter and concept key in one object).

JanKanis committed:
add EOL to _highdimComparison.gsp

gijskant committed:
Removed unneeded FIXME comment.

gijskant committed:
Changed concept name in view of SNP summary.

gijskant committed:
Improved performance of SNP autocomplete.

However, now the autocomplete is case sensitive.

gijskant committed:
Fixed SNP filter dialog, to properly initialise.

Because of the way the dialog is being reused
in different situations (data export,
summary statistics, grid view), the dialog needs
reinitialisation when opened in another place then where
it was created.

rnugraha committed:
Merge branch 'master' into snp-summary-grid

Conflicts:
	grails-app/services/com/recomdata/transmart/data/export/HighDimExportService.groovy
